### DISCUSSION TITLE
Supplying Source/Changelog URLs for Dependencies?

### DISCUSSION THREAD

**Discussion by straub** [renovatebot/renovate#14745] [Request Help] [answered]
In my homelab, I use (and use Renovate to manage), a few Docker images that aren't labeled with their source URL, so Renovate winds up making PRs with no link back to the source or changelog for the image. To work around this, when I'm reviewing these PRs, I'll typically google image name + "changelog" and find what I'm looking for in the first few results, but I'd love to be able to skip this after the first time and supply the handful of URLs via config.

Is there a way I can manually supply my own fallback source URL for a given dependency, such as in a package rule? I'm not finding anything that looks straightforward so far.

Alternatives I've considered:

- Overriding `prBodyDefinitions` `Package` in a package rule to make it link out to my chosen URL.
- Adding the source URL in an entry in `prBodyNotes` so the link is at least in there somewhere.

If it's not directly possible, is there an alternative you'd recommend, or that I haven't considered? And would anyone be interested in my attempting a PR to add a feature in this area (if that's even a reasonable idea)?

Thanks so much for the great tool and all your help!

**Comment by rarkins** [comment, marked as answer]
If running locally, I think you can use `packageRules` to specify `sourceUrl` for a package. It wasn't *intended* to be possible.. but I've heard it works. e.g.

```
"packageRules": [{
  "matchPackageNames": ["foo/bar"],
  "sourceUrl": "https://gitlab.company.com/foo/bar"
}]
```

**Reply by straub** [reply]
Ooh, I didn't realize that was actually possible. I'll give this a shot, thanks!

**Reply by straub** [reply]
Ah, at least using the GitHub app on my private repo, I got the following issue raised:
```
There is an error with this repository's Renovate configuration that needs to be fixed. As a precaution, Renovate will stop PRs until it is resolved.

Location: renovate.json
Error type: The renovate configuration file contains some invalid settings
Message: Invalid configuration option: packageRules[2].sourceUrl, Invalid configuration option: packageRules[3].sourceUrl, Invalid configuration option: packageRules[5].sourceUrl
```
I'll have to try again running locally a little later.

Would you folks be open to a PR that added more "official" support for this use case? I'd be willing to put in the leg work!

**Reply by rarkins** [reply]
Can you try putting it in your bot config, e.g. `config.js` and not in `renovate.json`. That way it bypasses the validity check

**Reply by viceice** [reply]
A shared preset will do too, that's what I use. 🙈🙈🙈

https://github.com/visualon/renovate-config/blob/main/meta.json

**Reply by straub** [reply]
Tested `config.js` and it worked like a charm! I got a `depNameLinked` and `references` in the template, just like I wanted.

Thanks for the shared preset reference too, I think that would make for an easier spot for me to manage overall.

I'm still up for taking a crack at a PR if you're open to it, but this will definitely work for me as-is. Thanks a ton!

**Comment by pxlfrk** [comment]
I'm curious - is there any option configuring this when running the Hosted Github App from Mend?
I tried adding the package-rule in my renovate.json, but got the same error-message as @straub :
https://github.com/renovatebot/renovate/discussions/14745#discussioncomment-2409392

**Reply by straub** [reply]
The shared preset option suggested by viceice worked for me while using the GitHub App: https://github.com/renovatebot/renovate/discussions/14745#discussioncomment-2410153
